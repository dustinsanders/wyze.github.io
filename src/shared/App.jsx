// @flow

import './styles/global'
import {
  Conclusion,
  CoreTeam,
  Employment,
  GitHubList,
  Introduction,
  Social,
} from './containers'
import type {
  GitHubResults,
  Jobs,
  Profiles,
  Style,
  Teams,
  VNode,
} from '../types'
import { css, media } from 'glamor'
import { large } from './styles/breakpoints'
import preact from 'preact'

type Props = {
  contributions: GitHubResults,
  jobs: Jobs,
  profiles: Profiles,
  projects: GitHubResults,
  teams: Teams,
}

const main: Style = css(
  {
    padding: '2em',
  },
  media(large, {
    padding: '4em 6em',
  }),
)

const App = ({
  contributions,
  jobs,
  profiles,
  projects,
  teams,
}: Props): VNode => (
  <main {...main}>
    <Introduction />
    <Social profiles={profiles} />
    <Employment jobs={jobs} />
    <CoreTeam teams={teams} />
    <GitHubList content={contributions} title="Contributions Made" />
    <GitHubList content={projects} title="Open Source Projects" />
    <Conclusion />
  </main>
)

export default App
