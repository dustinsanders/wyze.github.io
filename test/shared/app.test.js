// @flow

import App from '../../src/shared/App'
import snapshot from '../helpers/snapshot'

describe('<App />', () => {
  it('renders', snapshot.bind(null, App))
})
