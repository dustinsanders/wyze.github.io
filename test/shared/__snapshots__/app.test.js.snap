// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders 1`] = `
"<main data-css-qsx4ri>
  <Introduction></Introduction>
  <Social></Social>
  <Employment></Employment>
  <CoreTeam></CoreTeam>
  <GitHubList title=\\"Contributions Made\\"></GitHubList>
  <GitHubList title=\\"Open Source Projects\\"></GitHubList>
  <Conclusion></Conclusion>
</main>"
`;
